packages/number-input/src/Select/Select.tsx

```diff
@@ -0,0 +1,169 @@
+  } as const;
+
+  /**
+   * Gets the current unit option using the unit string
+   */
+  const currentUnitOption = unitOptions?.find(
+    unit => unit.displayName === unitProp,
+  );
+
+  const handleChange = (val: string) => {
+    const selectedUnit = unitOptions.find(unit => unit.displayName === val);
+
+    if (selectedUnit !== undefined) {
+      onChange(selectedUnit);
+    }
+  };
+
+  /**
+   * Custom unit button with a tooltip.
+   * Tooltip will show up if there is an ellipse.
+   */
+  const CustomMenuButton = React.forwardRef(
+    ({ className, children, ...props }: ButtonProps, forwardedRef) => {
+      const [open, setOpen] = useState<boolean>(false);
+      const buttonRef: React.MutableRefObject<HTMLElement | null> =
+        useForwardedRef(
+          forwardedRef,
+          null,
+        ) as React.MutableRefObject<HTMLElement | null>;
+
+      /**
+       * Gets the text node for the selected option.
+       */
+      const textNode = buttonRef.current?.querySelector(
+        `.${menuButtonTextClassName}`,
+      ) as HTMLElement;
+
+      /**
+       * Checks if the selected option has an ellipse.
+       */
+      const hasEllipse = textNode?.offsetWidth < textNode?.scrollWidth;
+
+      const handleMouseEnter = (e: React.MouseEvent<HTMLButtonElement>) => {
+        const popoverParent = (e.target as HTMLButtonElement).closest(
+          `.${popoverClassName}`,
+        );
+        if (!popoverParent) setOpen(true);
+      };
+
+      const handleMouseLeave = () => setOpen(false);
+      const handleOnFocus = () => setOpen(true);
+      const handleOnBlur = () => setOpen(false);
+
+      return (
+        <div className={customMenuButtonWrapperStyles}>
+          <Tooltip
+            enabled={hasEllipse && !disabled}
+            justify="middle"
+            // Using refEl instead of a trigger element because triggerProps by default, such as onMouseEnter, are added to the trigger element inside the tooltip component. OnMouseEnter is triggered by hovering over the trigger or any of its children. In the case of this custom menu button we don't want the tooltip to open when children are hovered so we add our own open logic with onMouseEnter.
+            refEl={buttonRef}
+            open={open}
+            {...popoverProps}
+          >
+            {currentUnitOption?.displayName}
+          </Tooltip>
+          <Button
+            {...props}
+            className={cx(
+              menuBaseStyles,
+              menuThemeStyles[theme],
+              {
+                [menuThemeDisabledStyles[theme]]: disabled,
+              },
+              className,
+            )}
+            ref={buttonRef}
+            onMouseEnter={handleMouseEnter}
+            onMouseLeave={handleMouseLeave}
+            onFocus={handleOnFocus}
+            onBlur={handleOnBlur}
+          >
+            {children}
+          </Button>
+        </div>
+      );
+    },
+  );
+
+  CustomMenuButton.displayName = 'CustomMenuButton';
```

Comment:
Yeah I'm feeling pretty strongly that should should be defined outside of the Select component?